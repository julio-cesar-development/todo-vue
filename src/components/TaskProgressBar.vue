<template>
  <div class="task-progress">
    <span class="progress-value">{{ progress.toFixed(2) }} %</span>
    <div class="progress-bar" v-bind:style="{ 'width': `${progress}%` }"></div>
  </div>
</template>

<script>
export default {
  name: 'TaskProgressBar',
  props: {
    progress: {
      type: Number,
      default: 0,
    }
  },
}
</script>

<style scoped>
.task-progress {
  box-sizing: border-box;
  position: relative;
  width: 80vw;
  height: 20px;
  border: 1px solid #fff;
  color: #fff;
  border-radius: 8px;
  margin-bottom: 30px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #fff2;
}

.progress-value {
  z-index: 1;
}

.progress-bar {
  position: absolute;
  border-radius: 8px;
  height: 18px;
  background-color: #4caf50;
  align-self: flex-start;
  transition: .5s;
}
</style>
