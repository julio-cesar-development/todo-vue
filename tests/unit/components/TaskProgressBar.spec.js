import { shallowMount } from '@vue/test-utils';
import TaskProgressBar from '@/components/TaskProgressBar.vue';

describe('TaskProgressBar.vue', () => {
  it('TaskProgressBar.vue should match all specifications bellow', () => {
    const props = {
      progress: 50,
    };
    const wrapper = shallowMount(TaskProgressBar, {
      'propsData': props,
    });
    expect(wrapper.isVueInstance()).toBeTruthy();

    expect(wrapper.classes('task-progress')).toBe(true);

    expect(wrapper.contains('span[class="progress-value"]')).toBe(true);

    const progressBar = wrapper.find('.progress-bar');
    expect(progressBar.html()).toEqual(`<div class="progress-bar" style="width: ${props.progress}%;"></div>`);

    expect(wrapper).toMatchSnapshot();
  });
});