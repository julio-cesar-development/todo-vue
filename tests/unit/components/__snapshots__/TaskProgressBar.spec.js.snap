// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskProgressBar.vue TaskProgressBar.vue should match all specifications bellow 1`] = `
<div class="task-progress"><span class="progress-value">50.00 %</span>
  <div class="progress-bar" style="width: 50%;"></div>
</div>
`;
